client/root/index.js

```diff
@@ -0,0 +1,14 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import { redirectRoot } from './controller';
+
+export default function() {
+	page( '/', redirectRoot );
+}
```

Comment:
Does this redirect handler really need to be a section? Could it instead be somewhere around [this code](https://github.com/Automattic/wp-calypso/blob/master/client/boot/common.js#L117-L130) which already handles special `/` redirects for desktop app or devdocs? (in logged out mode)

Reply:
Great point — no, it really doesn't have to be a separate section.

I think I was overthinking it, and imagining a future state where we're A/B testing redirecting to the customer home, and so mightn't want to have the server send the user the Reader chunks. But like you say below, that breaks an existing optimization, and your suggestion is much cleaner. I've updated this to remove the `root` section, and create a `loggedInMiddleware` for handling the redirect.